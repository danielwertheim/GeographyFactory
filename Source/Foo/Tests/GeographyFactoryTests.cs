using ApprovalTests;
using ApprovalTests.Reporters;
using NUnit.Framework;

namespace Foo.Tests
{
    [TestFixture]
    [UseReporter(typeof(DiffReporter))]
    public class GeographyFactoryTests
    {
        [Test]
        public virtual void When_Coordinates_has_correct_order_and_is_closed()
        {
            var g = GeographyFactory.CreatePolygon(CoordinatesTestFactory.CorrectOrderAndClosed());

            Approvals.Verify(g);
        }

        [Test]
        public virtual void When_Coordinates_has_correct_order_and_is_not_closed()
        {
            var g = GeographyFactory.CreatePolygon(CoordinatesTestFactory.CorrectOrderNotClosed());

            Approvals.Verify(g);
        }

        [Test]
        public virtual void When_Coordinates_has_wrong_order_and_closed()
        {
            var g = GeographyFactory.CreatePolygon(CoordinatesTestFactory.WrongOrderAndClosed());

            Approvals.Verify(g);
        }

        [Test]
        public virtual void When_Coordinates_has_wrong_order_and_not_closed()
        {
            var g = GeographyFactory.CreatePolygon(CoordinatesTestFactory.WrongOrderNotClosed());

            Approvals.Verify(g);
        }

        [Test]
        public virtual void When_Coordinates_has_wrong_order_starting_south_to_north_and_many_to_west_and_back_and_is_closed()
        {
            var g = GeographyFactory.CreatePolygon(CoordinatesTestFactory.WrongSouthToNorthAndManyWestAndBackClosed());

            Approvals.Verify(g);
        }

        [Test]
        public virtual void When_Castle()
        {
            var g = GeographyFactory.CreatePolygon(CoordinatesTestFactory.Castle());

            Approvals.Verify(g);
        }

        [Test]
        public virtual void When_Star_drawn_from_middle_and_clockwise()
        {
            var g = GeographyFactory.CreatePolygon(CoordinatesTestFactory.StartThatDoesNotRenderWithoutFix());

            Approvals.Verify(g);
        }
    }
}