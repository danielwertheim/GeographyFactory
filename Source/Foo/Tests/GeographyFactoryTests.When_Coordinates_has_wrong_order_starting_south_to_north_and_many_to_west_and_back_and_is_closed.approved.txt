POLYGON ((11.82043981552126 58.068356082112246, 11.819613695144673 58.06866818143402, 11.818626642227192 58.068549016560375, 11.81819748878481 58.068225567041509, 11.817489385604878 58.067675128668654, 11.81919527053835 58.067397068945283, 11.819205999374409 58.06806667848543, 11.82043981552126 58.068356082112246))